codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractDiskHttpData.java

```diff
@@ -156,17 +156,28 @@ public void addContent(ByteBuf buffer, boolean last)
                     throw new IOException("Out of size: " + (size + localsize) +
                             " > " + definedSize);
                 }
-                int written = 0;
                 if (file == null) {
                     file = tempFile();
                 }
                 if (fileChannel == null) {
                     RandomAccessFile accessFile = new RandomAccessFile(file, "rw");
                     fileChannel = accessFile.getChannel();
                 }
-                buffer.getBytes(buffer.readerIndex(), fileChannel, fileChannel.position(), localsize);
+                int totalWritten = 0;
```

Comment:
nit: would be tiny bit simpler to do decrement remaining instead

Reply:
@njhill sorry do you mean decrement `localsize` variable or something like this:
```
int remaining = localsize;
long position = fileChannel.position();
int index = buffer.readerIndex();
while (remaining > 0) {
   int written = buffer.getBytes(index, fileChannel, position, remaining);
   if (written < 0) {
      break;
   }
   remaining -= written;
   position += written;
   index += written;
}
fileChannel.position(position);
buffer.readerIndex(index);
size += localsize - remaining;
```

Reply:
@amizurov yes exactly, except to match the current logic the last line would still be just
```java
size += localsize;
```

But noticing that, it looks like this current (now merged) logic is actually slightly wrong and that (ignoring the `remaining` suggestion) this would be more correct:
```java
size += totalWritten;
```
i.e. to keep state consistent in the case where we exit prematurely due to `written < 0` (not sure it's realistic case but since we're handling it, should be done correctly)

WDYT? (cc @normanmaurer)

Reply:
@njhill yeah... Want to do a PR ?

Reply:
@njhill, @normanmaurer  I'll create separate PR with code above if you don't mind